crates/amaru-kernel/src/lib.rs

```diff
@@ -751,39 +752,36 @@ mod test {
-                // c4b115df82f104e71c173b4f73fd4efc03d87f4e81d6e8b0f2bb67b3ddc90235
-                cbor::decode(hex::decode("845820A6B764333F98CD288DA07626EBCF9D6C6D44C7BE73804030AE704821224D9E515840D0877153B391900CFB82FAEF05194F47D8793610EEDDA6B61C2DA451860E43329B159D4518FE93C9024829F4633C11CED09018258DA8CCC16AD33F9257AC3C0858205755653623B74D85A70B61A3F6B5F8D04701FCF80DBE3C4FADE7B6B554C8754B5825A201581E581C22C0BE3476E5BCE310D2B28B5D21DB7A536889A23C4D3E21F5B35016024101").expect("failed to decode bootstrap witness").as_slice()).expect("failed to decode bootstrap witness cbor"),
-                hex::decode("A5A8B29A838CE9525CE6C329C99DC89A31A7D8AE36A844EEF55D7EB9").expect("Failed to decode root")
-            ),
-            (
-                // 17d3497083063ef23d25280a64657a0efa0b53cc369cc5ef430941f5cc91c94e
-                cbor::decode(hex::decode("8458202B13CBA22AAD6A2B775FDA17F85E7BC74A5CEDB21DDE1D2BE14EC23BE2C9B3C2584006883DEDFFF9C6AF6B4422A81D4D6904C8372E4571D8610007473653722CC07841D5FFA7E427818DD6B9323DE629CF452C5EA08FEA90B2E33E577E26AD8BEF0F58209422C0D6AA61D186C40A0D562911311578463F91C26FA56F3FECD8BBEC0E3A1544A1024101").expect("failed to decode bootstrap witness").as_slice()).expect("failed to decode bootstrap witness cbor"),
-                hex::decode("59F44FD32EE319BCEA9A51E7B84D7C4CB86F7B9B12F337F6CA9E9C85").expect("Failed to decode root")
-            ),
-        ];
-
-        cases.into_iter().for_each(|(witness, expected_root)| {
-            assert_eq!(to_root(&witness).to_vec(), expected_root)
-        });
+    static BOOTSTRAP_WITNESSES: LazyLock<[(BootstrapWitness, Hash<28>); 5]> = LazyLock::new(|| {
+        macro_rules! import_bootstrap_witness {
+            ($hash:literal) => {
+                (
+                    from_cbor(include_bytes!(concat!(
+                        "../tests/data/bootstrap_witnesses/",
+                        $hash,
+                        ".cbor"
+                    )))
+                    .unwrap(),
+                    hash!($hash),
+                )
+            };
+        }
+
+        [
+            import_bootstrap_witness!("232b6238656c07529e08b152f669507e58e2cb7491d0b586d9dbe425"),
+            import_bootstrap_witness!("323ea3dd5b510b1bd5380b413477179df9a6de89027fd817207f32c6"),
+            import_bootstrap_witness!("59f44fd32ee319bcea9a51e7b84d7c4cb86f7b9b12f337f6ca9e9c85"),
+            import_bootstrap_witness!("65b1fe57f0ed455254aacf1486c448d7f34038c4c445fa905de33d8e"),
+            import_bootstrap_witness!("a5a8b29a838ce9525ce6c329c99dc89a31a7d8ae36a844eef55d7eb9"),
+        ]
+    });
+
+    #[test_case(&BOOTSTRAP_WITNESSES[0])]
+    #[test_case(&BOOTSTRAP_WITNESSES[1])]
+    #[test_case(&BOOTSTRAP_WITNESSES[2])]
+    #[test_case(&BOOTSTRAP_WITNESSES[3])]
+    #[test_case(&BOOTSTRAP_WITNESSES[4])]
+    fn to_root_key_hash((bootstrap_witness, root): &(BootstrapWitness, Hash<28>)) {
+        assert_eq!(to_root(bootstrap_witness).as_slice(), root.as_slice())
     }
```

Comment:
This now gives us one test per test case, which is (a) easier to read and understand what the test is doing, and (b) easier to troubleshoot in case the test fails since it directly identifies the failing scenario (c) in the long run, helps keep the test suite fast since independent test cases can run in parallel.